sdk/storage/inc/datalake/datalake_utilities.hpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "common/storage_url_builder.hpp"
+
+#include <map>
+#include <string>
+
+namespace Azure { namespace Storage { namespace DataLake {
+
+  namespace Details {
+    UrlBuilder GetBlobUriFromDfsUri(const UrlBuilder& dfsUri);
+  } // namespace Details
+
+  std::map<std::string, std::string> DeserializeMetadata(
+      const std::string& dataLakePropertiesString);
+
+  std::string SerializeMetadata(const std::map<std::string, std::string>& dataLakePropertiesMap);
```

Comment:
do we still need `DeserializeMetadata()` and `SerializeMetadata()`?
if so, why aren't they in Details namespace? Are users supposed to call them directly?

Reply:
I believe you are correct, we don't need those to be public anymore.

Reply:
In the first place, why do we still need this? I think you'll call blob APIs to set and get metadata.